bin/libs/component.sh

```diff
@@ -339,33 +339,23 @@ process_component_appfw_plugin() {
     # apply values if appfw_plugin_path has variables
     appfw_plugin_path=$(parse_string_vars "${appfw_plugin_path}")
     appfw_plugin_path=$(cd "${appfw_plugin_path}"; pwd)
-
-    if [ ! -r "${appfw_plugin_path}" ]; then
-      print_error "App Framework plugin directory ${appfw_plugin_path} is not accessible"
-      all_succeed=false
-      break
-    fi
     if [ ! -r "${appfw_plugin_path}/pluginDefinition.json" ]; then
-      print_error "App Framework plugin directory ${appfw_plugin_path} does not have pluginDefinition.json"
-      all_succeed=false
-      break
-    fi
-    appfw_plugin_id=$(read_json "${appfw_plugin_path}/pluginDefinition.json" ".identifier")
-    if [ -z "${appfw_plugin_id}" ]; then
-      print_error "Cannot read identifier from App Framework plugin ${appfw_plugin_path}/pluginDefinition.json"
+      print_error "App Framework plugin directory ${appfw_plugin_path} is not accessible or does not have pluginDefinition.json"
       all_succeed=false
       break
     fi
 
-    # copy to workspace/app-server/pluginDirs
-    appfw_plugin_workspace_path="${ZWE_zowe_workspaceDirectory}/app-server/pluginDirs/${appfw_plugin_id}"
-    mkdir -p "${appfw_plugin_workspace_path}"
-    cp -r "${appfw_plugin_path}/." "${appfw_plugin_workspace_path}/"
-
-    # install app
-    "${ZWE_zowe_runtimeDirectory}/components/app-server/share/zlux-app-server/bin/install-app.sh" "${appfw_plugin_workspace_path}"
-    # FIXME: do we know if install-app.sh fails. if so, we need to set all_succeed=false
+    if [ "${ZWE_RUN_ON_ZOS}" != "true" ]; then
+      # for containers, copy to workspace/app-server/pluginDirs and run install-app. on zos, this is done at startup.
+      appfw_plugin_workspace_path="${ZWE_zowe_workspaceDirectory}/app-server/pluginDirs/${appfw_plugin_id}"
```

Comment:
Seems the line of assigning value to `appfw_plugin_id` is removed. So it could be empty here.